framework/decode/dx12_browse_consumer.h

```diff
@@ -0,0 +1,353 @@
+        TrackTargetDrawcall(call_info, object_id);
+    }
+
+    virtual void Process_ID3D12GraphicsCommandList_Dispatch(const ApiCallInfo& call_info,
+                                                            format::HandleId   object_id,
+                                                            UINT               ThreadGroupCountX,
+                                                            UINT               ThreadGroupCountY,
+                                                            UINT               ThreadGroupCountZ)
+    {
+        TrackTargetDrawcall(call_info, object_id);
+    }
+
+    virtual void
+    Process_ID3D12CommandQueue_ExecuteCommandLists(const ApiCallInfo&                        call_info,
+                                                   format::HandleId                          object_id,
+                                                   UINT                                      NumCommandLists,
+                                                   HandlePointerDecoder<ID3D12CommandList*>* ppCommandLists)
+    {
+        if (target_command_list_ == format::kNullHandleId)
+        {
+            if (track_submit_index_ == dump_resources_target_.submit_index)
+            {
+                if (NumCommandLists <= dump_resources_target_.command_index)
+                {
+                    GFXRECON_LOG_FATAL("The target command index(%d) of dump resources is out of range(%d).",
+                                       dump_resources_target_.command_index,
+                                       NumCommandLists);
+                    GFXRECON_ASSERT(NumCommandLists > dump_resources_target_.command_index);
+                }
+                auto command_lists   = ppCommandLists->GetPointer();
+                target_command_list_ = command_lists[dump_resources_target_.command_index];
+
+                auto it = track_commandlist_infos_.find(target_command_list_);
+                GFXRECON_ASSERT(it != track_commandlist_infos_.end());
+
+                auto drawcall_size = it->second.track_dump_drawcalls.size();
+                if (drawcall_size <= dump_resources_target_.drawcall_index)
+                {
+                    GFXRECON_LOG_FATAL("The target drawcall index(%d) of dump resources is out of range(%d).",
+                                       dump_resources_target_.drawcall_index,
+                                       drawcall_size);
+                    GFXRECON_ASSERT(drawcall_size > dump_resources_target_.drawcall_index);
+                }
+                it->second.track_dump_drawcalls[dump_resources_target_.drawcall_index].execute_code_index =
+                    call_info.index;
+            }
+            ++track_submit_index_;
+        }
+    }
+
+    virtual bool IsComplete(uint64_t block_index) override
+    {
+        return (block_index > kMaxDX12BlockLimit) || (target_command_list_ != format::kNullHandleId);
+    }
+
+  private:
+    static int const    kMaxDX12BlockLimit = 1000;
```

Comment:
This will limit the `Dx12BrowseConsumer` to processing only the first 1000 GFXR blocks so it may not allow for dumping resources for draw calls that don't happen at the front of the file. There should probably not be a block limit when computing `IsComplete` for this consumer.

Reply:
Thank you. Fixed.